Tests/Invoke-PxGet.Tests.ps1

```diff
@@ -0,0 +1,200 @@
+
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+BeforeAll {
+    $script:testRoot  = $null
+    $script:moduleList = @()
+    $script:failed = $false
+    $script:testNum = 0
+
+    & (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-Test.ps1' -Resolve)
+
+    function Init
+    {
+        $script:testRoot = $null
+        $script:moduleList = @()
+        $script:failed = $false
+        $Global:Error.Clear()
+    }
+    
+    function GivenPxGetFile
+    {
+        param(
+            [Parameter(Mandatory)]
+            [string] $Contents
+        )
+
+        New-Item -Path $testRoot  -ItemType 'File' -Name "pxget.json" -Value $Contents
+    }
+
+    function New-TestRoot
+    {
+        $script:testRoot = Join-Path -Path $TestDrive -ChildPath ($script:testNum++)
+        New-Item -Path $script:testRoot -ItemType 'Directory'
+    }
+
+    function RemovePxGetFile
+    {
+        if( Test-Path -Path "$testRoot \pxget.json" )
+        {
+            Remove-Item -Path "$testRoot \pxget.json"
+        }
+    }
+    
+    function Reset
+    {
+        RemovePxGetFile
+    }
+
+    function ThenErrorThrown
+    {
+        param(
+            [Parameter(Mandatory)]
+            [string] $WithError
+        )
+    
+        $Global:Error[-1] | Should -Match $WithError
+    }
+    
+    function ThenSucceeded
+    {
+        $script:failed | Should -BeFalse
+        $Global:Error | Should -BeNullOrEmpty
+        Assert-MockCalled -CommandName 'Get-RootDirectory' -ModuleName 'PxGet' -Times 2
+    }
+    
+    function WhenInvokingPxGet
+    {
+        try 
+        {
+            Mock -CommandName 'Get-RootDirectory' -ModuleName 'PxGet' { return $testRoot}
+            Invoke-PxGet 'install'
+        }
+        catch 
+        {
+            $script:failed = $true
+            Write-Error -ErrorRecord $_ -ErrorAction $ErrorActionPreference
+        }
+    }
+}
+
+Describe 'Invoke-Pxget' {
+    BeforeEach { 
+        Init
+        New-TestRoot 
+    }
+    AfterEach { 
+        Reset 
+    }
+
+    It 'should pass when there is a valid module in the pxget file' {
+        $contents = @"
+        {
+            "PSModules": [
+                {
+                    "Name": "PackageManagement",

```

Comment:
Update all the tests to install the `NoOp` module instead. https://www.powershellgallery.com/packages/NoOp/1.0.0